### PR TITLE
Added invariants for each of the contracts in the client/application directory

### PR DESCRIPTION
**Pull request by mpetersoCode55**
Added invariants for each of the contracts in the client/application directory

### CODE REVIEW

File: docs/userGuides/development/invariants/client/application/AppManager-Invariants.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+- If removePauseRule is called by an account that is not a Rule Admin the transaction will be reverted.
+- If activatePauseRuleCheck is called by an account that is not a Rule Admin the transaction will be reverted.
+
+## Tags Invariants
+
+- If addTag is called by an account that is not an App Admin the transaction will be reverted.
+- If addTagToMultipleAccounts is called by an account that is not an App Admin the transaction will be reverted.
+- If addMultipleTagToMultipleAccounts is called by an account that is not an App Admin the transaction will be reverted.
+- If removeTag is called by an account that is not an App Admin the transaction will be reverted.
+
```
Comment (TJ-Everett) [inline docs/userGuides/development/invariants/client/application/AppManager-Invariants.md:54]: I think we should have the event emission for each of the data contract functions.

File: docs/userGuides/development/invariants/client/application/ProtocolApplicationHandler-Invariants.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# ApplicationProtocolHandler Invariants
+
+- Upon creation of the contract the App Manager address passed in to the constructor will be the contracts owner.
+- If checkApplicationRules is called by an account that is not the registered AppManager the transaction will be reverted.
+
```
Comment (TJ-Everett) [inline docs/userGuides/development/invariants/client/application/ProtocolApplicationHandler-Invariants.md:5]: In Shane's current invariant ticket he has this: 
When AccountMinMaxTokenBalance is active, violation of the rule is reverted.
- When AccountMaxBuySize is active, violation of the rule is reverted. 
- When TokenMaxBuyVolume is active, violation of the rule is reverted. 
- When AccountMaxSellSize is active, violation of the rule is reverted. 
- When TokenMaxSellVolume is active, violation of the rule is reverted. 

see: https://github.com/thrackle-io/tron/pull/270/files#diff-6b0e19a2db0415a3e097542eb1f12cbab623dd0409528bb6e9e1a7cfc32ec47b

I think this is necessary here too for the app level rules

**Review by TJ-Everett — CHANGES_REQUESTED** [review]
Great work over all TDMike! Just a quick couple things about event emission and the rules reverting (example given).

File: docs/userGuides/development/invariants/client/application/ProtocolApplicationHandler-Invariants.md (hunk 2 above)
Comment (mpetersoCode55) [reply docs/userGuides/development/invariants/client/application/ProtocolApplicationHandler-Invariants.md:5]: Good catch! I shall make it so

File: docs/userGuides/development/invariants/client/application/AppManager-Invariants.md (hunk 1 above)
Comment (mpetersoCode55) [reply docs/userGuides/development/invariants/client/application/AppManager-Invariants.md:54]: That's embarrassing, I can't believe I missed those. On it!

File: docs/userGuides/development/invariants/client/application/ProtocolApplicationHandler-Invariants.md (hunk 2 above)
Comment (mpetersoCode55) [reply docs/userGuides/development/invariants/client/application/ProtocolApplicationHandler-Invariants.md:5]: Resolved

File: docs/userGuides/development/invariants/client/application/AppManager-Invariants.md (hunk 1 above)
Comment (mpetersoCode55) [reply docs/userGuides/development/invariants/client/application/AppManager-Invariants.md:54]: Resolved

**Review by TJ-Everett — APPROVED** [review]
Looks great! Approved

**Review by ShaneDuncan602 — APPROVED** [review]
This looks great. Tayler already found all the good stuff!

**Pull request merged**